packages/core/src/internal/clients/leaderboard/AbstractLeaderboard.ts

```diff
@@ -89,28 +88,31 @@ export abstract class AbstractLeaderboard implements ILeaderboard {
    * Note: can fetch a maximum of 8192 elements at a time and rank
    * is 0-based (index begins at 0).
    *
-   * @param {LeaderboardFetchByRankOptions} options
-   * @param {number} [options.startRank] - The rank of the first element to
-   * fetch. Defaults to 0. This rank is inclusive, ie the element at this rank
-   * will be fetched.
-   * @param {number} [options.endRank] - The rank of the last element to fetch.
+   * @param {number} [startRank] - The rank of the first element to
+   * fetch. This rank is inclusive, ie the element at this rank
+   * will be fetched. Ranks can be used to manually paginate through the leaderboard
+   * in batches of 8192 elements (e.g. request 0-8192, then 8192-16384, etc).
+   * @param {number} [endRank] - The rank of the last element to fetch.
    * This rank is exclusive, ie the element at this rank will not be fetched.
-   * Defaults to startRank + 8192 in order to fetch the maximum 8192 elements per request.
+   * Ranks can be used to manually paginate through the leaderboard
+   * in batches of 8192 elements (e.g. request 0-8192, then 8192-16384, etc).
+   * @param {LeaderboardFetchByRankOptions} options
    * @param {LeaderboardOrder} [options.order] - The order to fetch the elements in.
    * Defaults to ascending, meaning 0 is the lowest-scoring rank.
    * @returns {Promise<LeaderboardFetch.Response>} -
-   * {@link LeaderboardFetch.Found} containing the requested elements.
-   * {@link LeaderboardFetch.NotFound} when requested elements were not found.
+   * {@link LeaderboardFetch.Success} containing the requested elements.
    * {@link LeaderboardFetch.Error} on failure.
    */
-  public async leaderboardFetchByRank(
+  public async fetchByRank(
+    startRank: number,
+    endRank: number,
```

Comment:
I hadn't thought about how making these required would cause us to pull them up to positional arguments, or else make the `options` argument required.

What you did here makes sense, but the one downside is that because JS doesn't allow callers to use keyword-style call patterns for positional arguments, you can no longer say `startRank: blah` in your call to these, it'll just be `fetchByRank(12, 42)`, which isn't quite as readable.

but it is more succinct. and it seems fairly intuitive. so i think i'm on board with doing it this way, vs. keeping these inside the options interface and just making the options arg required.

@bruuuuuuuce I am curious if you have any opinion on this.

Reply:
:skeletor_fist_shake: nodejs 😠 . 

Generally speaking, I like to use objects as params to my functions in js for this exact reason. However,

- this is more concise
- this is the pattern we currently have set for our other methods
- most js sdks just use positional args like this [like the redis sdk](https://github.com/redis/node-redis#basic-example)

I think its ok to ship with this